scripts/export-signupless-reportbacks-to-rogue.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Script to export Reportbacks that do not have a corresponding Signup from drupal into Rogue.
+ *
+ * to run:
+ * drush --script-path=../scripts/ php-script export-signupless-reportbacks-to-rogue.php
+ */
+
+// Get all the signupless reportbacks
+$reportbacks = db_query("SELECT rb.rbid, rb.nid, rb.run_nid, rb.quantity, rb.why_participated, rb.rbid, rb.flagged, rb.uid
+  FROM dosomething.dosomething_reportback rb
+  LEFT JOIN dosomething.dosomething_signup ds on rb.uid = ds.uid AND rb.run_nid = ds.run_nid
+  WHERE ds.sid IS NULL
+  AND rb.rbid NOT IN (SELECT rbid from dosomething_rogue_reportbacks)");
+
+$client = dosomething_rogue_client();
+
+if (!$reportbacks) {
+  echo 'Nothing to migrate!';
+}
+
+// For each reportback, send a request for each reportback file to POST /posts and Rogue will automatically create the signup
+foreach ($reportbacks as $reportback) {
+  echo 'On rbid ' . $reportback->rbid . '...' . PHP_EOL;
+  $sent_at = date('Y-m-d H:i:s');
+
+  // Try to get Northstar ID
+  $northstar_id = dosomething_user_get_northstar_id($reportback->uid, 'drupal_id');
+
+  if (!$northstar_id) {
+    echo "\t" . 'No northstar id, that is terrible ' . $reportback->rbid . PHP_EOL;
+
+    continue;
```

Comment:
do we log the reportback anywhere (in a failed logs table maybe?) that doesn't have a northstar id to try to re-send/investigate later?

Reply:
@chloealee we do not, the thinking is that if there is no user there's not much else we can do. These ones will be tried every time we run the script though, and we can see in the logs which ones don't have northstar ids if we want to

Reply:
ah sorry! I see in 120 the logic behind this that whenever we run the script these will always be tried!

Reply:
@chloealee yup that's exactly right!